lib/iris/fileformats/__init__.py

```diff
@@ -74,10 +77,11 @@ def _pp_little_endian(filename, *args, **kwargs):
 #
 # NB. Because this is such a "fuzzy" check, we give this a very low
 # priority to avoid collateral damage from false positives.
-FORMAT_AGENT.add_spec(
-    FormatSpecification('GRIB', MagicNumber(100),
-                        lambda header_bytes: b'GRIB' in header_bytes,
-                        grib.load_cubes, priority=1))
+if grib is not None:
+    FORMAT_AGENT.add_spec(
+        FormatSpecification('GRIB', MagicNumber(100),
+                            lambda header_bytes: b'GRIB' in header_bytes,
+                            grib.load_cubes, priority=1))
```

Comment:
Nice if this still knew what a grib file was, and told us we didn't have grib installed.